nntrainer/layers/cl_layers/fc_layer_cl.cpp

```diff
@@ -0,0 +1,616 @@
+      1, &inputX, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemv.SetKernelArguments(
+      2, &inOutY, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemv.SetKernelArguments(
+      3, &dim1, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemv.SetKernelArguments(
+      4, &dim2, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    const int work_groups_count[3] = {(int)dim1, 1, 1};
+    const int work_group_size[3] = {32, 32, 1}; // test-value
+
+    result = context.command_queue_inst_.DispatchCommand(
+      FullyConnectedLayerCl::kernel_sgemv, work_groups_count, work_group_size);
+    if (!result) {
+      break;
+    }
+
+    result = inOutY.ReadData(context.command_queue_inst_, vecYdata);
+    if (!result) {
+      break;
+    }
+
+  } while (false);
+}
+
+float FullyConnectedLayerCl::fc_dot_cl(const float *matAdata,
+                                       const float *vecXdata, unsigned int dim1,
+                                       RunLayerContext &context) {
+
+  bool result = false;
+
+  float cl_ret = 0;
+
+  do {
+    // FullyConnectedLayerCl::kernel_ is wrong for this ...its sgemv.
+    result =
+      context.clCreateKernel(fc_dot_cl_kernel_, context.LayerKernel::FCDOT,
+                             FullyConnectedLayerCl::kernel_dot);
+    if (!result) {
+      break;
+    }
+
+    size_t dim1_size = sizeof(float) * dim1;
+
+    opencl::Buffer inputA(context.context_inst_, dim1_size, true, nullptr);
+
+    opencl::Buffer inputX(context.context_inst_, dim1_size, true, nullptr);
+
+    result = inputA.WriteData(context.command_queue_inst_, matAdata);
+    if (!result) {
+      break;
+    }
+
+    result = inputX.WriteData(context.command_queue_inst_, vecXdata);
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_dot.SetKernelArguments(
+      0, &inputA, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_dot.SetKernelArguments(
+      1, &inputX, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_dot.SetKernelArguments(2, &dim1,
+                                                                  sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_dot.SetKernelArguments(
+      3, &cl_ret, sizeof(float));
+    if (!result) {
+      break;
+    }
+
+    const int work_groups_count[3] = {(int)dim1, 1, 1};
+    const int work_group_size[3] = {32, 32, 1}; // test-value
+
+    result = context.command_queue_inst_.DispatchCommand(
+      FullyConnectedLayerCl::kernel_dot, work_groups_count, work_group_size);
+    if (!result) {
+      break;
+    }
+
+  } while (false);
+
+  return cl_ret;
+}
+
+void FullyConnectedLayerCl::fc_sgemm_cl(const float *A, const float *B,
```

Comment:
I wonder if we can use these blas cl operations in other layers?

Reply:
Added  in latest commit